packages/web-frontend/src/sagas.js

```diff
@@ -884,12 +886,32 @@ function* fetchMeasureInfo(measureId) {
       ? measureParams
       : getDefaultDates(measureParams);
 
+  // This param should default to false, used when large quantity
+  // of entities is detrimental to performance. Consequence is more fetching.
+  // displayOnLevel: SubDistrict
+  const { fetchOnlyDisplayLevel = false } = measureParams;
+  const measureDataAncestor = fetchOnlyDisplayLevel
+    ? selectDisplayLevelAncestor(state)
+    : countryCode;
+
+  const shouldFetchCurrentLevel = selectShouldFetchCurrentLevel(state);
+
+  if (!shouldFetchCurrentLevel) {
+    // Don't try and fetch null measures
+    yield put(cancelFetchMeasureData());
```

Comment:
The behaviour of this feels a bit buggy. For instance, if we select a regularly map overlay, and then switch to one where the `displayOnLevel` is lower than the current org unit, it just leaves the previous map overlay showing...

I've tried a couple quick things to see if this is easily remedied, but I think it's a bit tricky... We basically need to yield an action that sets the currently selected measure in the state properly, but doesn't actually fetch data...

Happy to chat on this if ya like 👍